### ISSUE TITLE
Deep point reference implementation

### ISSUE THREAD

**Issue by UlisesGascon** [GoblinDBRocks/GoblinDB#17]
Support to deep points like, "user/data/etc...". Need to valite and react with creation.

**Comment by CodingCarlos** [comment]
Created [**deep-references**, a new branch from dev](https://github.com/GoblinDBRocks/GoblinDB/compare/dev...deep-references) and added basic working for DB (24cfc24f629af10d8ba4b5e2afe3e4aef6e11a0c). I've not added test coverage yet, and I've not seen if shall I do the same for Ambush functions. That's why I've created a new branch.

To make it work, I've settled as separator a dot `.` as is the Javascript default character to define an object property.

For example:

```goblin.set('demo', 'inner.object.name');```

will set goblin.db to something like this:

```
{
    inner: {
        object: {
            name: 'demo'
        }
    }
}
```

It's also working for get, update and push methods, in the same way.

We can make it work also with slashes `/` instead of (or added to) dots, if you prefer, just by checking before doing the split if there are dots or slashes in the point.

Also, I know the code is not great, but a little bit dirty. It's the first approach, and I know I shall refactor and clean it, but having #18 open, I've been more relaxed on the way to write code.
